Mage.Sets/src/mage/cards/a/Anathemancer.java

```diff
@@ -61,8 +62,10 @@ public int calculate(Game game, Ability sourceAbility, Effect effect) {
         if (sourceAbility.getFirstTarget() == null) {
             return 0;
         }
+        FilterLandPermanent filter = FilterLandPermanent.nonbasicLands();
+        filter.add(new ControllerIdPredicate(sourceAbility.getFirstTarget()));
         return game.getBattlefield().count(
-                StaticFilters.FILTER_LANDS_NONBASIC,
+                filter,
                 sourceAbility.getControllerId(),
```

Comment:
another way to do this is using `FilterControlledLandPermanent` with a nonbasic predicate and then passing in `sourceAbility.getFirstTarget()` to the `count` function